### PR TITLE
TOF: change fetching of conditions from the ccdb

### PR DESCRIPTION
**Pull request by sofiatomassini**
the url to get the reference map in the ccdb is now set correctly. 
@Barthelemy  could you please merge this PR? Thank you!

### CODE REVIEW

**Comment by njacazio** [comment]
@Barthelemy we were wondering if this is how we should set up the CCDB call to customize the URL. Thanks and cheers,
N.

**Review by Barthelemy — CHANGES_REQUESTED** [review]
Hi, 

I understand that you want to retrieve something from the CCDB ? 

There are several ways of doing it. You should not set the url yourself. The URL is set in the config file. 

The documentation is here: 
https://github.com/AliceO2Group/QualityControl/blob/master/doc/Advanced.md#accessing-objects-in-ccdb

The recommended way is to use Lifetime::Condition DPL input. There are many examples in the Modules. 

We don't recommend but don't forbid either to use `retrieveConditionAny` which would let you retrieve the CCDB data as well. It is ready to use and does not require any extra configuration.

**Comment by knopers8** [comment]
PostProcessingTasks do not take DPL inputs, so in this case `retrieveConditionAny` should be used.

**Comment by sofiatomassini** [comment]
Indeed, however the `PostProcessingInterface.h` does not depend on `UserCodeInterface.h` where the `retrieveConditionAny` is defined. For instance in `Modules/CPV/src/PhysicsTask.cxx` it can be used  because the `TaskInterface.h` is dependent on `UserCodeInterface.h`. Did I understand correctly how It works? Thank you !

**Comment by knopers8** [comment]
PostProcessingInterface does inherit UserCodeInterface since two months: https://github.com/AliceO2Group/QualityControl/blob/07bdae6d68188960dd14c5be555d149c4bd17ee9/Framework/include/QualityControl/PostProcessingInterface.h#L37
Perhaps you have to rebase with the master branch?

**Comment by Barthelemy** [comment]
I am updating the documentation accordingly

File: Modules/TOF/include/Base/MessagePad.h [hunk 1]
```diff
@@ -35,13 +35,13 @@ namespace o2::quality_control_modules::tof
 
 struct MessagePad {
   float mPadLowX = 0.6;                 /// Position of the message PAD in low x
-  float mPadLowY = 0.5;                 /// Position of the message PAD in low y
```
Comment (njacazio) [inline Modules/TOF/include/Base/MessagePad.h:38]: don't change these

File: Modules/TOF/include/Base/MessagePad.h [hunk 2]
```diff
@@ -140,7 +140,7 @@ struct MessagePad {
     ILOG(Info, Devel) << "Message pad '" << mName << "' is enabled" << ENDM;
 
     mMessagePad = new TPaveText(mPadLowX, mPadLowY, mPadHighX, mPadHighY, padOpt);
-    mMessagePad->SetName(Form("%s_msg", histogram->GetName()));
```
Comment (njacazio) [inline Modules/TOF/include/Base/MessagePad.h:143]: this is ok

File: Modules/TOF/include/Base/MessagePad.h [hunk 3]
```diff
@@ -35,13 +35,13 @@ namespace o2::quality_control_modules::tof
...
-  float mPadLowY = 0.5;                 /// Position of the message PAD in low y
+  float mPadLowY = 0.9;                 /// Position of the message PAD in low y
   float mPadHighX = 0.9;                /// Position of the message PAD in high x
-  float mPadHighY = 0.75;               /// Position of the message PAD in high y
+  float mPadHighY = 1.5;                /// Position of the message PAD in high y
   std::vector<std::string> mMessages{}; /// Message to print on the pad, this is reset at each call of MakeMessagePad
   TPaveText* mMessagePad = nullptr;     /// Text pad with the messages
   int mEnabledFlag = 1;                 /// Flag to enable or disable the pad
-  const std::string mName = "";         /// Name of the message pad, can be used to identify the pad if multiple are used
+  const std::string mName = "msg";      /// Name of the message pad, can be used to identify the pad if multiple are used
```
Comment (njacazio) [inline Modules/TOF/include/Base/MessagePad.h:44]: this is ok

File: Modules/TOF/src/PostProcessHitMap.cxx [hunk 4]
```diff
@@ -60,22 +63,24 @@ void PostProcessHitMap::initialize(Trigger, framework::ServiceRegistryRef servic
 {
   // Setting up services
   mDatabase = &services.get<o2::quality_control::repository::DatabaseInterface>();
+  o2::ccdb::BasicCCDBManager::instance().setURL(mCCDBUrl);
 }
 
 void PostProcessHitMap::update(Trigger t, framework::ServiceRegistryRef services)
 {
   // Getting the hit map
-  const auto mo = mDatabase->retrieveMO(mCCDBPath, mCCDBPathObject, t.timestamp, t.activity);
```
Comment (njacazio) [inline Modules/TOF/src/PostProcessHitMap.cxx:68]: Don't change this

File: Modules/TOF/src/PostProcessHitMap.cxx [hunk 5]
```diff
@@ -60,22 +63,24 @@ void PostProcessHitMap::initialize(Trigger, framework::ServiceRegistryRef servic
...
     return;
   }
   // Getting the reference map
-  auto& ccdb = services.get<repository::DatabaseInterface>();
-  map<string, string> metadata; // can be empty
-  const auto* refmap = static_cast<o2::tof::TOFFEElightInfo*>(ccdb.retrieveAny(typeid(o2::tof::TOFFEElightInfo), mRefMapCcdbPath, metadata, mRefMapTimestamp));
+  const o2::tof::TOFFEElightInfo* refmap = nullptr;
+  refmap = o2::quality_control::core::UserCodeInterface::retrieveConditionAny<o2::tof::TOFFEElightInfo>(mRefMapCcdbPath);
 
+  // const auto* refmap = o2::ccdb::BasicCCDBManager::instance().getForTimeStamp<o2::tof::TOFFEElightInfo>(mRefMapCcdbPath, mRefMapTimestamp);
```
Comment (njacazio) [inline Modules/TOF/src/PostProcessHitMap.cxx:82]: This can be cleaned

File: Modules/TOF/src/PostProcessHitMap.cxx [hunk 6]
```diff
@@ -24,7 +24,10 @@
 #include "QualityControl/MonitorObject.h"
 #include "QualityControl/QcInfoLogger.h"
 #include "TOF/PostProcessHitMap.h"
+#include "CCDB/CcdbApi.h"
```
Comment (njacazio) [inline Modules/TOF/src/PostProcessHitMap.cxx:27]: These new includes are not needed then, right?

File: Modules/TOF/src/PostProcessHitMap.cxx [hunk 7]
```diff
@@ -46,8 +49,8 @@ void PostProcessHitMap::configure(const boost::property_tree::ptree& config)
   const std::string baseJsonPath = "qc.postprocessing." + getID() + ".customization.";
   mCCDBPath = config.get<std::string>(baseJsonPath + "CCDBPath", "TOF/MO/TaskDigits/");
   ILOG(Info, Support) << "Setting CCDBPath to " << mCCDBPath << ENDM;
-  mCCDBPathObject = config.get<std::string>(baseJsonPath + "CCDBPathObject", "HitMapNoiseFiltered");
```
Comment (njacazio) [inline Modules/TOF/src/PostProcessHitMap.cxx:49]: Keep both

File: Modules/TOF/src/PostProcessHitMap.cxx [hunk 8]
```diff
@@ -60,22 +63,24 @@ void PostProcessHitMap::initialize(Trigger, framework::ServiceRegistryRef servic
 {
   // Setting up services
   mDatabase = &services.get<o2::quality_control::repository::DatabaseInterface>();
+  o2::ccdb::BasicCCDBManager::instance().setURL(mCCDBUrl);
```
Comment (njacazio) [inline Modules/TOF/src/PostProcessHitMap.cxx:66]: This might not be needed

**Review by njacazio — CHANGES_REQUESTED** [review]

**Review by njacazio — APPROVED** [review]

**Comment by njacazio** [comment]
Ok @Barthelemy it should be good to merge

**Comment by Barthelemy** [comment]
why do you need `mCCDBUrl` ?

**Comment by njacazio** [comment]
Indeed it was not needed

**Review by Barthelemy — APPROVED** [review]

**Review by njacazio — APPROVED** [review]

**Pull request merged**